src/actions/channels.js

```diff
@@ -582,7 +582,11 @@ export function joinChannel(userId, teamId, channelId, channelName) {
                 channel = await Client4.getChannel(channelId);
             } else if (channelName) {
                 channel = await Client4.getChannelByName(teamId, channelName);
-                member = await Client4.addToChannel(userId, channel.id);
+                if ((channel.type === General.GM_CHANNEL) || (channel.type === General.DM_CHANNEL)) {
+                    member = await Client4.getChannelMember(channel.id, userId);
```

Comment:
If the group or dm does not exist then the channel member does not exist either, im not sure this works

Reply:
Exactly. Currently the group or channel must exist, if not it raise an error, which is captured later.

The current behavior is raise an error (in the addToChannel) always when you try to join an existing or not existing DM/GM channel.

The full workflow is, send a `/groupmsg` or `/msg` which creates the DM or GM, and return a GotoLocation to redirect to the channel. Because the channel, doesn't exist in the store, It try to join to the channel, and because is a group or msg, it fails. With this PR it joins because expect when you join to a DM o GM you must already be a member, if not, fails anyway.

Reply:
Makes sense